### PR TITLE
Clean code of shared folder

### PR DESCRIPTION
**Pull request by xpwmaosldk**
- Change local variables type (ref: https://dart.dev/guides/language/effective-dart/design#dont-redundantly-type-annotate-initialized-local-variables)
- Move UploadProfileImage method from UserService to NetworkService

### CODE REVIEW

File: mobile/lib/shared/views/immich_loading_overlay.dart [hunk 1]
```diff
@@ -9,25 +9,25 @@ class ImmichLoadingOverlay extends StatelessWidget {
...
-          return Container();
-        }
+        return shouldShow
+            ? const Scaffold(
+                backgroundColor: Colors.black54,
+                body: Center(
+                  child: ImmichLoadingIndicator(),
+                ),
+              )
+            : const SizedBox();
```
Comment (alextran1502) [inline mobile/lib/shared/views/immich_loading_overlay.dart:22]: I am not used to return `SizedBox` for empty Widget, is it the same as return `Container()`?

File: mobile/lib/shared/services/network.service.dart [hunk 2]
```diff
@@ -4,15 +4,22 @@ import 'dart:convert';
...
+import 'package:image_picker/image_picker.dart';
 import 'package:immich_mobile/constants/hive_box.dart';
 import 'package:immich_mobile/utils/dio_http_interceptor.dart';
+import 'package:immich_mobile/utils/files_helper.dart';
 
 class NetworkService {
+  late final Dio dio;
+
+  NetworkService() {
+    dio = Dio();
```
Comment (alextran1502) [inline mobile/lib/shared/services/network.service.dart:17]: This is very cool, to instantiate only once and add the interceptor at the constructor. Love this change!

**Comment by alextran1502** [comment]
This is good to merge, let me know if there is anything else you would like to add

File: mobile/lib/shared/views/immich_loading_overlay.dart (hunk 1 above)
Comment (alextran1502) [reply mobile/lib/shared/views/immich_loading_overlay.dart:22]: Learning a lot from you!

File: mobile/lib/shared/services/network.service.dart [hunk 3]
```diff
@@ -145,4 +136,34 @@ class NetworkService {
       return false;
     }
   }
+
+  Future<Response?> uploadProfileImage(XFile image) async {
```
[the remark this answers is not in the document]
Comment (xpwmaosldk) [reply mobile/lib/shared/services/network.service.dart:140]: In my opinion, only the parts necessary to manipulate user data other than the network should be executed here.

NetworkService is already used to get user information here

[the remark this answers is not in the document]
Comment (alextran1502) [reply mobile/lib/shared/services/network.service.dart:140]: Hmm then I would make the method something more generic like upload file(s) with optional headers sending to the server

[the remark this answers is not in the document]
Comment (xpwmaosldk) [reply mobile/lib/shared/services/network.service.dart:140]: Okay.
I will put this change on hold until we get a better design. :P

**Pull request merged**